src/passes/DeadArgumentElimination.cpp

```diff
@@ -615,6 +615,10 @@ struct DAE : public Pass {
   // function subtyping in indirect calls. TODO: relax this when possible
   //
   // Returns whether we optimized.
+  //
+  // TODO: We may be missing a global optimum here, as if a function calls
+  //       itself and returns that value, then we would not do any change here,
+  //       as one of the return values is exactly what it already is.
```

Comment:
It may be worth noting that the recursion could be mutual with some set of other functions as well, rather than just direct.